frontend/src/stores/auth.js

```diff
@@ -0,0 +1,172 @@
+import { defineStore } from "pinia";
+import { Notify } from "quasar";
+
+const baseUrl = `${process.env.API}`;
+
+export const useAuthStore = defineStore("auth", {
+  state: () => ({
+    user: JSON.parse(localStorage.getItem("user")) || null,
+    token: localStorage.getItem("token") || null,
+    refreshTokenTimeout: localStorage.getItem("refreshTokenTimeout") || null,
+  }),
+
+  getters: {
+    current_user(state) {
+      return state.user;
+    },
+  },
+
+  actions: {
+    async login(email, password) {
+      const formData = new URLSearchParams();
+      formData.append("grant_type", "password");
+      formData.append("username", email);
+      formData.append("password", password);
+      try {
+        const response = await fetch(`${baseUrl}/token`, {
+          method: "POST",
+          headers: {
+            Accept: "application/json",
+            "Content-Type": "application/x-www-form-urlencoded",
+          },
+          body: formData.toString(),
+        });
+
+        // Check for HTTP errors
+        if (!response.ok) {
+          // Extract error message if available
+          const errorData = await response.json().catch(() => ({})); // Parsing might fail, default to empty object
+          const errorMessage =
+            errorData.detail || `Error: ${response.statusText}`;
```

Comment:
If the user is not authenticated, proper error message is not displayed.  It always says "Error: Not Found"